libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+import * as pathUtilities from 'path';
+import * as fs from 'fs';
+import * as fsx from 'fs-extra';
+
+import { Text } from './';
+
+/**
+ * The allowed types of encodings, as supported by Node.js
+ * @public
+ */
+export type Encoding = 'utf8';
+
+/**
+ * Available file handle opening flags.
+ * @public
+ */
+export type FileHandleOpenFlags = 'r' | 'r+' | 'rs+' | 'w' | 'wx' | 'w+' | 'wx+' | 'a' | 'ax' | 'a+' | 'ax+';
+
+/**
+ * Enumeration controlling conversion of newline characters.
+ * @public
+ */
+export enum NewlineConversion {
+  CrLf,
+  Lf,
+  None
+}
+
+/**
+ * The options for FileSystem.readFolder()
+ * @public
+ */
+export interface IReadFolderOptions {
+  /**
+   * If true, returns the absolute paths of the files in the folder.
+   * Defaults to `false`.
+   */
+  absolutePath?: boolean;
+}
+
+/**
+ * The options for FileSystem.writeFile()
+ * @public
+ */
+export interface IWriteFileOptions {
+  /**
+   * If true, will ensure the folder is created before writing the file.
+   * Defaults to `false`.
+   */
+  ensureFolder?: boolean;
+
+  /**
+   * If specified, will normalize line endings to the specified style of newline.
+   * Defaults to `NewlineConversion.None`.
+   */
+  convertLineEndings?: NewlineConversion;
+
+  /**
+   * If specified, will change the encoding of the file that will be written.
+   * Defaults to `"utf8"`.
+   */
+  encoding?: Encoding;
+}
+
+/**
+ * The options for FileSystem.readFile()
+ * @public
+ */
+export interface IReadFileOptions {
+  /**
+   * If specified, will change the encoding of the file that will be written.
+   * Defaults to `"utf8"`.
+   */
+  encoding?: Encoding;
+
+  /**
+   * If specified, will normalize line endings to the specified style of newline.
+   * Defaults to `NewlineConversion.None`.
+   */
+  convertLineEndings?: NewlineConversion;
+}
+
+/**
+ * The options for FileSystem.move()
+ * @public
+ */
+export interface IMoveOptions {
+  /**
+   * If true, will overwrite the file if it already exists. Defaults to false.
+   */
+  overwrite?: boolean;
+
+  /**
+   * If true, will ensure the folder is created before writing the file.
+   * Defaults to `false`.
+   */
+  ensureFolder?: boolean;
+}
+
+/**
+ * The options for FileSystem.deleteFile()
+ * @public
+*/
+export interface IDeleteFileOptions {
+  /**
+   * If true, will throw an exception if the file did not exist before `deleteFile()` was called.
+   * Defaults to `false`.
+   */
+  throwIfNotExists?: boolean;
+}
+
+/**
+ * A standardized file system API. Mostly contains wrappers around `fs` and `fs-extra`.
+ * Also contains useful options to simplify and normalize file system operations.
+ * @public
+ */
+export class FileSystem {
+
+  // ===============
+  // COMMON OPERATIONS
+  // ===============
+
+  /**
+   * Returns true if the path exists on disk. It can be a file, folder, or link.
```

Comment:
>It [](start = 46, length = 2)

The path can refer to a

Reply:
maybe define "filesystem object" in the class comments

---
In reply to: [198664455](https://github.com/Microsoft/web-build-tools/pull/727#discussion_r198664455) [](ancestors = 198664455)